bioblend/galaxy/invocations/__init__.py

```diff
@@ -11,9 +11,33 @@ def __init__(self, galaxy_instance):
         self.module = 'invocations'
         super().__init__(galaxy_instance)
 
-    def get_invocations(self):
+    def get_invocations(self, workflow_id=None, history_id=None, user_id=None,
+                        include_terminal=True, limit=None, view='collection',
+                        step_details=False):
         """
-        Get a list containing all workflow invocations.
+        Get a list containing all the workflow invocations corresponding to the
+        specified workflow.
+
+        :type workflow_id: str
+        :param workflow_id: Encoded workflow ID to restrict the query.
+
+        :type history_id: str
+        :param history_id: Encoded history ID to restrict the query.
+
+        :type user_id: str
+        :param user_id: Encoded user ID to restrict the query. This must be
+                        your own user ID if your are not an admin user.
+
+        :type limit: int
+-       :param limit: Maximum number of invocations in the query result.
```

Comment:
It seems so.

Reply:
It seems to be the first `limit` invocations ordered by their `id`.

Query: https://github.com/galaxyproject/galaxy/blob/85fafd5a0628b8e127c111c84ce2263678dc220a/lib/galaxy/managers/workflows.py#L253

Limit applied: https://github.com/galaxyproject/galaxy/blob/85fafd5a0628b8e127c111c84ce2263678dc220a/lib/galaxy/managers/workflows.py#L281

Reply:
If you try `/api/invocations` it's clearly ordered by time though and `/api/invocations?limit=5` shows the most recent ones. Not sure what to make of the code you linked.

Reply:
I looked at the index_invocations() of galaxy and what it does with the limit parameter. If the output is ordered by time then I misinterpreted what it does

Reply:
Yeah, I don't really understand what the code is doing there - as you say it looks like it sorts by ID

Reply:
Ordering by id or create time should be equivalent, since it's order on the autoincrement IDs stored in the database, not the encoded IDs returned by the API, or am I missing something?

Reply:
No, that makes perfect sense.